### PR TITLE
Part #19737 - Replace deprecated FormField component with TextField, Label, HelpText

### LINKED ISSUE #19737: Replace deprecated FormField component with FormTextField from the component-library
### Description

Currently, the extension is using an outdated `FormField` component, which needs to be replaced with the new `FormTextField` component.

This is a massive undertaking by itself and creating a single PR would be too large. Smaller PRs can be submitted against this issue to ensure easier review and gradual improvements.

### Technical Details

- Replace instances of FormField component (`ui/components/ui/form-field/form-field.js`) with FormTextField component (`ui/components/component-library/form-text-field/form-text-field.js`)
- Component APIs are slightly different so ensure all props have been migrated appropriately

### Acceptance Criteria

- Instances of FormField component are completely replaced with the new FormTextField component
- The component APIs are updated to reflect the changes in the new FormField component and there is no functional changes or visual regression
- Each Pull Request (PR) should include **no more than 3 files**
- The code changes should pass Jest tests, linting, and Storybook without any errors.
- The PR must include before and after screenshots of the UI to ensure there are no visual regressions.

If the acceptance criteria is not met, PRs may be closed.

### Difficulty: Intermediate

Good first issue for: External contributors who are familiar with running the extension locally, have knowledge of React, component props, Jest tests, linting, and Storybook, and want to contribute to improving the cohesiveness of UI in the extension

### PR DESCRIPTION
**Pull request by Jainex17**
## Explanation

This pull request addresses the Issue of replacing the deprecated `FormField` component with the modernized `Label`, `TextField` and `HelpText` components from our component library. Additionally, it includes updates to the constants previously used in the codebase, changing `DISPLAY` to `Display`, `FLEX_DIRECTION` to `FlexDirection`, and `BLOCK_SIZES` to `BlockSize`.

* Part #19737 

## Changes made
* Replaced instances of the deprecated `FormField` component with `TextField`, `Label` and `HelpText` components.
* Updated constants: `DISPLAY` to `Display`, `FLEX_DIRECTION` to `FlexDirection`, and `BLOCK_SIZES` to `BlockSize`.

### Before
Some of the icons and text were unaligned and inconsistent. The input type is also text when it is a number input 

https://github.com/MetaMask/metamask-extension/assets/8112138/aef8a0d1-cbed-4a92-9431-d985020f1dad

Storybook also didn't work as expected you can't edit the input

https://github.com/MetaMask/metamask-extension/assets/8112138/289c3e96-48ac-484e-bad2-4ba91558ae39

### After
Icons are now consistent in the label and tooltip, the alignment of the max button is centered the label has also been updated to align with the design system

https://github.com/MetaMask/metamask-extension/assets/8112138/8a500fdb-e394-4463-82a6-98ed521db3d6

Storybook now works as expected being able to use the input to enter numbers

https://github.com/MetaMask/metamask-extension/assets/8112138/c120fd5c-6ce2-46ca-a515-ac11331dce4d

## Manual Testing Steps
Storybook
- Pull this branch
- Serach `CustomSpendingCap` in stroybook
- Check against `develop` storybook https://metamask.github.io/metamask-storybook/?path=/story/components-app-customspendingcap--custom-spending-cap-story&args=dappProposedValue:234234234

Check custom spending cap in app
- Go to test dapp https://metamask.github.io/test-dapp/
- Click `CREATE TOKEN` you will need test eth
- Wait for transaction
- Click `APPROVE TOKENS` 
- Use Custom Spending Cap component 

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

### CODE REVIEW

**Comment by Jainex17** [comment]
Hello @georgewrmarshall 

I've submitted a pull request with several changes and improvements to the codebase. However, there are currently three failing tests in the PR:

* ci/circleci: test-e2e-chrome — Failed on CircleCI
* ci/circleci: test-e2e-firefox — Failed on CircleCI
* ci/circleci: test-unit-jest-main

If you have the time would you be able to lend a hand in resolving these failing tests? Your expertise would be greatly appreciated. However, if the issues are complex or if you are occupied with other tasks, I'm completely open to closing this PR and working on a different issue. Please let me know your preferred course of action.

Thank you

**Comment by georgewrmarshall** [comment]
Hey @Jainex17, any chance you could rebase and try again? I will look into the e2e test failures once that's done

**Comment by Jainex17** [comment]
Hey @georgewrmarshall, I've pulled the latest changes and addressed the issues from scratch. I've also made some modifications and drop code in resolve option both js and snap files. However that tests are still failing sorry for that.🥲

# screenshots
## Before


## After

**Comment by georgewrmarshall** [comment]
Hey @Jainex17, Would it be possible to update this PR?

**Comment by Jainex17** [comment]
Hey @georgewrmarshall, I resolved the conflict. Could you please review it?

**Comment by Jainex17** [comment]
> Hey @Jainex17, Would it be possible to update this PR?

Hey @georgewrmarshall,  I have made updates to some of the code to fix the Jest errors, and while some of them have been resolved successfully, a few still remain. I have attempted to address those remaining issues but, unfortunately, have not been able to resolve them.

If you have the time, could you please take a look at the remaining error? I have attached a screenshot after making some changes.

Thank you for your assistance.

Screenshot 2023-07-26 133939

File: ui/components/app/custom-spending-cap/custom-spending-cap.js [hunk 1]
```diff
@@ -231,16 +238,26 @@ export default function CustomSpendingCap({
...
+              marginTop={1}
+            >
+              <Box
+                display={Display.Flex}
+                alignItems={AlignItems.center}
+                gap={2}
+              >
+                <Label
+                  htmlFor="custom-spending-cap"
+                  fontWeight={FontWeight.Bold}
```
Comment (georgewrmarshall) [inline ui/components/app/custom-spending-cap/custom-spending-cap.js:255]: I think we can remove the fontWeight here
```suggestion
```

Comment (Jainex17) [reply ui/components/app/custom-spending-cap/custom-spending-cap.js:255]: > I think we can remove the fontWeight here

Removing fontWeight may cause a slight difference from the original font.

I have attached both the `before` and `after` screenshots for your review. If everything appears satisfactory, we can proceed with the changes.

<table>
<tr>
<td>original</td>
<td>before remove</td>
<td>After remove</td>
</tr>
<tr>

<td> </td>

<td>  </td>

<td>  </td>

</tr>
</table>

**Comment by Jainex17** [comment]
Hello I've addressed the lint error, but some Jest tests are still failing. Could you please take a look?

File: ui/components/app/custom-spending-cap/custom-spending-cap.js (hunk 1 above)
Comment (georgewrmarshall) [reply ui/components/app/custom-spending-cap/custom-spending-cap.js:259]: Great spotting @Jainex17 the design team has update the `Label` to use medium font weight so we can align them here

File: test/e2e/tests/custom-token-add-approve.spec.js [hunk 2]
```diff
@@ -376,7 +376,7 @@ describe('Create token, approve token and approve token without gas', function (
 
         // set max spending cap
         await driver.clickElement({
-          css: '.custom-spending-cap__max',
+          css: '[data-testid="custom-spending-cap-max-button"]',
```
Comment (georgewrmarshall) [inline test/e2e/tests/custom-token-add-approve.spec.js:379]: Updating to use data-testid

File: ui/components/app/app-components.scss [hunk 3]
```diff
@@ -20,7 +20,6 @@
 @import 'connected-accounts-permissions/index';
 @import 'connected-sites-list/index';
 @import 'create-new-vault/create-new-vault.scss';
-@import 'custom-spending-cap/index';
```
Comment (georgewrmarshall) [inline ui/components/app/app-components.scss:23]: Removing unused styles

**Review by garrettbear — APPROVED** [review]
LGTM

**Review by georgewrmarshall — APPROVED** [review]
LGTM

**Pull request merged**